### PR TITLE
RIA-5798: Changed for new event EndAppealAutomatically

### PR DESCRIPTION
**Pull request by sharmad-v1**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForRemissionRejectedTrigger.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.clients.model.TimedEvent;
+
+@Component
+public class AutomaticEndAppealForRemissionRejectedTrigger implements PostSubmitCallbackHandler<AsylumCase> {
+
+    private final DateProvider dateProvider;
+    private final Scheduler scheduler;
+
+    @Value("${paymentAfterRemissionRejection.dueInMinutes}")
+    int schedule14DaysInMinutes;
```
Comment (sajjadh-v1) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForRemissionRejectedTrigger.java:28]: private?

**Review by sajjadh-v1 — DISMISSED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForRemissionRejectedTrigger.java (hunk 1 above)
Comment (sharmad-v1) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForRemissionRejectedTrigger.java:28]: Scope needed is to the present class only.

Comment (sajjadh-v1) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForRemissionRejectedTrigger.java:28]: For @Value, private member with a setter injection (@Setter Lombok) is a better practice, keeping Class' strongly encapsulated

**Comment by Marcellofabbri** [comment]
I've added some details to this PR, in the AutomaticEndAppealForRemissionRejectedTrigger class.

The condition for triggering the scheduling of the 'endAppealAutomatically' event was:
```
remissionDecision.isPresent()
&& remissionDecision.get() == RemissionDecision.REJECTED
```

I changed it to
```
callback.getEvent() == Event.RECORD_REMISSION_DECISION
&& remissionDecision.isPresent()
&& remissionDecision.get() == RemissionDecision.REJECTED
```

So that the 'endAppealAutomatically' event scheduling only happens when the admin uses the 'recordRemissionDecision' event to reject the remission (otherwise I think it would have triggered it at the end of whatever event after remission is rejected.

**Review by Marcellofabbri — APPROVED** [review]

**Review by Ryan-Nesbitt — APPROVED** [review]

**Pull request merged**